keps/sig-contributor-experience/0007-20180403-community-forum.md

```diff
@@ -0,0 +1,183 @@
+
+### Non-Goals
+
+- This is not a proposal to replace Slack, this is a proposal for a community forum.
+  - The motivation of having searchable information that is owned by the Kubernetes community comes from voiced concerns about having so much of Kubernetes depend on Slack. 
+  - You are encouraged to propose a KEP for real-time communication if you would like to champion that, this KEP is not about Slack. 
+- This does not replace Stack Overflow or kubernetes-users for user support. 
+  - However inevitably users who prefer forums will undoubtedly use it for support.
+  - Strictly policing "this should be posted here, that should be posted there" won't work. 
+  - I believe our community is large enough where we can have a support section, and there are enough people to make that self sustaining, we can also encourage cross posting from StackOverflow to integrate things better as both sites have good integration points.
+  - Over time as community interaction and knowledge base builds people will end up with a better experience than in #kubernetes-users on slack and will naturally gravitate there.
+  - Other large OSS communities have a presence on both StackOverflow and do support on their user forums already and it doesn't appear to be a big issue.
+- This will not replace kubernetes-devel or SIG mailing lists.
+  - They work, but we could experiment with mailing list integration. (See below)
+  - Let's concentrate on an end-user experience for now, and allow SIGs and working groups who want a more user-facing experience to opt-in if they wish. 
+
+## Proposal
+
+### User Stories
+
+(This is a bullet list right now, needs to be fleshed out)
+
+- A place for open ended discussion. For example "What CI/CD tools is everyone using?"
+  - This would be closed as offtopic on StackOverflow, but would be perfect for a forum.
+- Post announcements about kubernetes core that are important for end users
+  - An announcements subforum can be connected to Slack so that we have a single place for us to post announcements that get's propagated to other services.
+- Post announcements about related kubernetes projects
+  - Give the ecosystem of tools around k8s a place to go and build communities around all the tools people are building. 
+  - "Jill's neat K8s project on github" is too small to have it's own official k8s presence, but it could be a post on a forum. 
+- Events section for meetups and Kubecon
+- Sub boards for meetup groups
+- Sub boards for non-english speaking community members
+- Developer section can include:
+  - Updated posting of community meeting notes
+  - We can inline the youtube videos as well 
+  - Steering Committee announcements
+  - Link to important SIG announcements
+  - Any related user-facing announcements for our mentorship programs
+- Job board
+  - This might be difficult to do properly and keep classy, but leaving it here as a discussion point.
```

Comment:
I'd be in favor of a moderated job board, and willing to help moderate it.

Reply:
The main reason to moderate it is because otherwise we'll get bot-based spamming of the board.

Reply:
We should add more bots and spam and thus more moderation hours to the drawbacks section @castrojo

Reply:
Moderation is handled by a community trust system, what happens is we pick the 5 or so admins, and over time trusted users get more moderation powers. I don't expect that there will be extra moderation for people not engaged in the forum.